api/v1alpha1/types_etcd.go

```diff
@@ -485,3 +567,48 @@ func (e *Etcd) GetRoleName() string {
 func (e *Etcd) GetRoleBindingName() string {
 	return fmt.Sprintf("%s:etcd:%s", GroupVersion.Group, e.Name)
 }
+
+// IsBackupStoreEnabled returns true if backup store has been enabled for this etcd, else returns false.
+func (e *Etcd) IsBackupStoreEnabled() bool {
+	return e.Spec.Backup.Store != nil
+}
+
+// IsMarkedForDeletion returns true if a deletion timestamp has been set and false otherwise.
+func (e *Etcd) IsMarkedForDeletion() bool {
+	return !e.DeletionTimestamp.IsZero()
+}
+
+// GetSuspendEtcdSpecReconcileAnnotationKey gets the annotation key set on an etcd resource signalling the intent
+// to suspend spec reconciliation for this etcd resource. If no annotation is set then it will return nil.
+func (e *Etcd) GetSuspendEtcdSpecReconcileAnnotationKey() *string {
+	if metav1.HasAnnotation(e.ObjectMeta, SuspendEtcdSpecReconcileAnnotation) {
+		return pointer.String(SuspendEtcdSpecReconcileAnnotation)
+	}
+	if metav1.HasAnnotation(e.ObjectMeta, IgnoreReconciliationAnnotation) {
```

Comment:
When the annotation itself would go away in the future, all code that has to do with this field will be touched as well. Since we have already marked the annotation as deprecated, we need not add another `TODO` here. If we do, then we'll have to add a `TODO` everywhere the annotation is referenced, which isn't very nice.